index.bs

```diff
@@ -230,36 +243,37 @@ partial dictionary XRSessionInit {
 
 <div class="algorithm" data-algorithm="find-supported-configuration">
-In order to <dfn>find supported configuration combination</dfn> for depth sensing API given |usagePreference| and |dataFormatPreference| sequences, the user agent MUST run the following algorithm:
+In order to <dfn>find supported configuration combination</dfn> for depth sensing API given |depthStateInit| dictionary, the user agent MUST run the following algorithm:
 
+  1. Let |depthTypeRequest| be the value contained by the {{XRDepthStateInit/depthTypeRequest}} key in |depthStateInit| if it is set, and an empty sequence otherwise.
+  1. Let |selectedType| be <code>null</code>
+  1. Let |usagePreference| be the value contained by the {{XRDepthStateInit/usagePreference}} key in |depthStateInit|
   1. Let |selectedUsage| be <code>null</code>.
-  1. Initialize |selectedUsage| as follows:
-    <dl class="switch">
-      <dt class="switch">If the |usagePreference| sequence is empty
-        <dd> Initialize |selectedUsage| to the device's [=preferred native depth sensing capability=].
-      <dt> Otherwise
-        <dd>  For each |usage| in |usagePreference| sequence, perform the following steps:
-          1. If |usage| is not considered a <a lt="support depth sensing usage">supported depth sensing usage</a> by the [=native depth sensing=] capabilities of the device, continue to the next entry.
-          1. Set |selectedUsage| to |usage| and abort these nested steps.
-      </dl>
-  1. If |selectedUsage| is <code>null</code>, return <code>null</code> and abort these steps.
+  1. Let |dataFormatPreference| be the value contained by the {{XRDepthStateInit/dataFormatPreference}} key in |depthStateInit|
   1. Let |selectedDataFormat| be <code>null</code>.
-  1. Initialize |selectedDataFormat| as follows:
-    <dl class="switch">
-      <dt class="switch">If the |usagePreference| sequence is empty
-        <dd> Initialize |selectedDataFormat| to the device's [=preferred native depth sensing format=].
-      <dt> Otherwise
-        <dd>  For each |dataFormat| in |dataFormatPreference|, perform the following steps:
-          1. If |selectedUsage|,|dataFormat| is not considered a <a lt="support depth sensing usage and data format combination">supported depth sensng usage and data format combination</a> by the [=native depth sensing=] capabilities of the device, continue to the next entry.
-          1. Set |selectedDataFormat| to |dataFormat| and abort these nested steps.
-      </dl>
-  1. If |selectedDataFormat| is <code>null</code>, return <code>null</code> and abort these steps.
-  1. Return |selectedUsage|,|selectedDataFormat|.
+  1. Let |processingOrder| be the sequence of (preferences, selection) pairs, where selection is a reference: [(|depthTypeRequest|, |selectedType|), (|usagePreference|,|selectedUsage|),(|dataFormatPreference|,|selectedDataFormat|)]
```

Comment:
Nit: s/"selection is a reference"/"selection is a reference to one of the variables introduced in previous steps".